src/FrameSemantics.cc

```diff
@@ -574,128 +796,36 @@ Errors buildFrameAttachedToGraph(ScopedGraph<FrameAttachedToGraph> &_out,
   edge.SetWeight(0);
 
   // add link vertices
-  for (const auto &link : _model->Links())
-  {
-    if (outModel.Count(link.Name()) > 0)
-    {
-      errors.push_back({ErrorCode::DUPLICATE_NAME,
-          "Link with non-unique name [" + link.Name() +
-          "] detected in model with name [" + _model->Name() +
-          "]."});
-      continue;
-    }
-    outModel.AddVertex(link.Name(), sdf::FrameType::LINK);
-  }
+  addVerticesToGraph(outModel, _model.links, _model, errors);
 
-  // add joint vertices
-  for (const auto &joint : _model->Joints())
-  {
-    if (outModel.Count(joint.Name()) > 0)
-    {
-      errors.push_back({ErrorCode::DUPLICATE_NAME,
-          "Joint with non-unique name [" + joint.Name() +
-          "] detected in model with name [" + _model->Name() +
-          "]."});
-      continue;
-    }
-    outModel.AddVertex(joint.Name(), sdf::FrameType::JOINT);
-  }
+  // add interface joint vertices
```

Comment:
`_model.joints` contain both joints and interface joints correct? If so,
```suggestion
  // add joint vertices
```

Reply:
That's correct. Changed in https://github.com/ignitionrobotics/sdformat/pull/764/commits/59050f96134d6bf0cea07e35ab7916b1da7b70d9.